jsx/Breadcrumbs.js

```diff
@@ -93,33 +96,37 @@ window.removeEventListener('resize', this.checkScreenSize);
 
     if (dropdown.length !== 0) {
       breadcrumbDropdown = (
-        <div className="btn-group ellipsis btn btn-primary">
-          <button type="button"
-                  className="dropdown-toggle"
-                  data-toggle="dropdown"
-                  aria-haspopup="true"
-                  aria-expanded="false"
+        <div className='btn-group ellipsis btn btn-primary'>
+          <button type='button'
+                  className='dropdown-toggle'
+                  data-toggle='dropdown'
+                  aria-haspopup='true'
+                  aria-expanded='false'
           >
             ...
           </button>
-          <ul className="dropdown-menu">
+          <ul className='dropdown-menu'>
             {dropdown}
           </ul>
         </div>
       );
     }
 
     return (
-      <div id="bc2" className="btn-group btn-breadcrumb">
-        <a href={baseURL} className="btn btn-primary">
-          <span className="glyphicon glyphicon-home" aria-hidden="true"></span>
+      <div id='bc2' className='btn-group btn-breadcrumb'>
+        <a href={baseURL} className='btn btn-primary'>
+          <span className='glyphicon glyphicon-home' aria-hidden='true'/>
         </a>
         {breadcrumbDropdown}
         {breadcrumbs}
       </div>
     );
   }
 }
+Breadcrumbs.propTypes = {
+  baseURL: PropTypes.string,
+  breadcrumbs: PropTypes.array,
+};
 
 let RBreadcrumbs = React.createFactory(Breadcrumbs);
```

Comment:
This appears to be legacy syntax and should likely be removed throughout the code.

https://reactjs.org/docs/react-api.html#createfactory

Reply:
i agree. i think we can use export instead?

Reply:
@zaliqarosli @HenriRabalais I'm having trouble with updating this one. Not sure what to replace it with because after the changes suggested it will not render the breadcrumbs. Thoughts?

Reply:
@zaliqarosli and @HenriRabalais It turns out the main.tpl is only capable of using the RBreadcrumbs without JSX. So I'll leave it for another PR that cleans up the main.tpl

Reply:
How about for the other components?